lib/filters.rb

```diff
@@ -175,7 +175,7 @@ def self.merge_timewindows(vrp)
 
       service.activity.timewindows.each{ |timewindow|
         unified_timewindows[timewindow.id] = {
-          start: timewindow.start && ((timewindow.day_index || 0) * 86400 + timewindow.start) || (0 + (timewindow.day_index || 0) * 86400),
+          start: (timewindow.day_index || 0) * 86400 + timewindow.start,
```

Comment:
I ask myself if we can do a similar thing for `(timewindow.day_index || 0)` as well or not?

Reply:
I don't think we can because timewindow.day_index == nil can mean that timewindow is valid every day. 

Let's have : 
- vehicle1.timewindow = { start: 0, end: 1000, day_index: 0 } and vehicle2.timewindow = { start: 0, end: 1000, day_index: 2 },
- service.activity.timewindows = [{ start: 5, end: 10 }]

If we set day_index = 0 (by default) to the service then problem changes, it is like service can only take place on mondays which does not correspond to initial constraint.
 
If we do want to always have day_index for timewindows then in create we should duplicate timewindows with no day_index in order to have one timewindow per day_index. I do not think it is worth to do that.